main.js

```diff
@@ -0,0 +1,132 @@
+  
+  // 3. Написати функцію, яка приймає 1 аргумент та перевіряє чи дане число ідеальне
+  const PERFECT_NUMBER = function (num) {
+    let step = 0;
+    for (let i = 1; i <= num/2; i++) {
+        if (num % i === 0) {
+            step += i;
+        }
+    }
+    if (num === undefined || num == null) {
+        return "Not enough digits!"
+    } else if (arguments.length > 1) {
+        return "You have to put only 1 digit, not more!";
+    } else if ((typeof num !== 'number')) {
+        return "You have to put only digits!";
+    } else if (step === num && step !== 0) {
+        return `The number ${num} is perfect!`;
+    } else {
+        return `The number ${num} is NOT perfect!`;
+    }
+  }
+  console.log(PERFECT_NUMBER(6)) // The number 6 is perfect!
+  //console.log(PERFECT_NUMBER(5)) // The number 5 is NOT perfect!
+  //console.log(PERFECT_NUMBER(5, 7)) // You have to put only 1 digit, not more!
+  //console.log(PERFECT_NUMBER()) // Not enough digits!
+  //console.log(PERFECT_NUMBER("g")) // You have to put only digits!
+
+  // 4. Написати функцію, яка приймає 2 аргументи та обраховує суму в цьому проміжку. Зверніть увагу, що можна передати 10 та -2, як аргументи. Суму всіх чисел в цьому проміжку також треба обчислити
+    const RANGE_SUM = function(num1, num2) {
+    let sum = 0;
+    if (num1 === undefined || num2 === undefined || num1 == null || num2 == null) {
+        return "Not enough digits!"
+    } else if ((typeof num1 !== 'number') || (typeof num2 !== 'number')) {
+        return "You have to put only digits!";
+    } else if (arguments.length > 2) {
+        return "You have to put only 2 digits, not more!";
+    } else if (num2 < num1) {
+        for (let i = num2; i <= num1; i++) {
+            sum += i;
+        } 
+        return `Суму всіх чисел в проміжку ${num1} - ${num2} = ${sum}`;
+    } else {
+        for (let i = num1; i <= num2; i++) {
+            sum += i;
+        } return `Сумма всіх чисел в проміжку від ${num1} до ${num2} = ${sum}`;
+    }
+}
+console.log(RANGE_SUM(2, 6)) // Сумма всіх чисел в проміжку від 2 до 6 = 20
+// console.log(RANGE_SUM(-2, 6)) // Сумма всіх чисел в проміжку від -2 до 6 = 18
+// console.log(RANGE_SUM(2, -6)) // Суму всіх чисел в проміжку 2 - -6 = -18
+// console.log(RANGE_SUM(2, 6, 9)) // You have to put only 2 digits, not more!
+// console.log(RANGE_SUM()) // Not enough digits!
+// console.log(RANGE_SUM(2, "j")) // You have to put only digits!
+
+// 5. Написати функцію, яка приймає та конвертує температуру із Цельсій у Фаренгейт
+const TEMP_F = function (randomTemp) {
+    if (randomTemp === undefined || randomTemp == null) {
+        return "Please put a temperature to convert!"
+    } else if ((typeof randomTemp !== 'number')) {
+        return "Something went wrong :( We can't do the operation...";
+    }
+    const CONVERTATION_FROM_C_TO_F = randomTemp * 1.8 + 32;
+    return `${randomTemp}C = ${CONVERTATION_FROM_C_TO_F}F`
+}
+console.log(TEMP_F(100)) // 100C = 212 F
```

Comment:
<img width="464" alt="Screenshot 2022-09-20 at 21 33 50" src="https://user-images.githubusercontent.com/34886027/191337112-0628d0c9-f952-45bd-b003-c00f2425b2ac.png">

We can let this function calculate if the string can be converted to `Number`. Use it this way 
`if (!isNaN(+randomTemp))`